### PR TITLE
OCPBUGS-42045: Add IBM Public Cloud PreSubmit E2E for Ingress Operator

### PR DESCRIPTION
**Pull request by gcs278**
The Ingress Operator was lacking a presubmit E2E job for IBM Public Cloud. This change adds a IBM Public Cloud presubmit job as optional and always_run=false so that we can identify issues in CI.

### CODE REVIEW

**Comment by gcs278** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
Not that bad [first run](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56785/rehearse-56785-pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator/1835769103659307008), looks like the issue is Load Balancer HostName resolution. It's taking over 5 minutes for IBM Public Cloud. I think it's related to https://issues.redhat.com/browse/OCPBUGS-14966

**Comment by gcs278** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
CI infra issue:

/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

File: ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -1107,6 +1107,85 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-hypershift,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (Miciah) [inline ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml:1111]: Can this be `always_run: false`, at least initially?

**Comment by gcs278** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
Not sure if this is a glitch:
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
I think `always_run` isn't valid in the config files, let me try again with it just in the job:
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
Never mind, `always_run` is valid config, not sure why my pj rehearse job is failing:
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
I wonder if I hit a CI bug:
```
{  failed to generate steps from config: failed to get steps from configuration: failed to read buildRootImageStream from repository: failed to read .ci-operator.yaml file: open /home/prow/go/src/github.com/openshift/release/.ci-operator.yaml: no such file or directory}
```
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
Reverting my `always_run` change to see if that caused the CI issue.

**Comment by gcs278** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
Let me try to close and reopen.
/close

**Comment by gcs278** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
:crossed_fingers: I think it's working now.

**Comment by lihongan** [comment]
/pj-rehearse pull-ci-openshift-cluster-ingress-operator-master-e2e-ibmcloud-operator

**Comment by gcs278** [comment]
I ack the failures in the new `e2e-ibmcloud-operator` to be DNS propagation issues, which either we will be able to resolve either via DNS configuration changes (not sure if that's possible with IBM Cloud), altering the E2E tests (like extending the timeouts), or by skipping the E2E tests altogether.

The test is `always_run: false` and `optional: true` to ensure we can merge without creating an E2E blockage. I need this test to merge so that I can reopen https://github.com/openshift/cluster-ingress-operator/pull/1132 to provide data on how long DNS propagation is taking, or whether it's working at all.

/pj-rehearse ack

@miciah requesting another review please

**Comment by Miciah** [comment]
Should we also be updating `ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-release-4.18-presubmits.yaml` and `ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-release-4.19-presubmits.yaml`, or are those handled by a separate reconciliation job?

**Comment by gcs278** [comment]
> Should we also be updating `ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-release-4.18-presubmits.yaml` and `ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-release-4.19-presubmits.yaml`, or are those handled by a separate reconciliation job?

Though I don't think it hurts, I also don't think we have to manually maintain those files, if I understand this correctly: https://docs.ci.openshift.org/docs/architecture/branching/#do-i-need-to-make-sure-branched-ci-operator-configs-are-up-to-date-for-release-branches.
```
No. When the branch is cut and goes live, the current CI Operator configuration for the development branch is used to seed the configuration for the release branch. These are not DRY intentionally to allow for drift between CI for branches.
```

And examining the `Automated Config Brancher` that @openshift-bot is working here: https://github.com/openshift/release/pulls?q=is%3Apr++is%3Aclosed+author%3Aopenshift-bot+, it seems like it might currently reconcile the 4.18 & 4.19 release branches on a very routine basis. 

Taking an real example, Candace added the `e2e-gateway-api` in https://github.com/openshift/release/pull/48873 during the 4.16 development cycle, but shortly after the openshift-ci bot merged [this](https://github.com/openshift/release/pull/50828/files#diff-ac5ee6188f88b0714acb332dd35c829b4acdc1f0f752d660ebf0297e441ee213) which updated 4.16 & 4.17. So TIL we don't need to manage the release-in-development (or > release-in-development) ci-operator branched files.

**Comment by Miciah** [comment]
Thanks!
/lgtm

**Comment by gcs278** [comment]
@Miciah I think a recent update caused `ci/prow/ordered-prow-config` to fail and required me to run `make jobs` again. I also rebased to be safe.

Could I get another LGTM please?

**Comment by gcs278** [comment]
Whoops, did `make jobs`, then rebase, and needed to do another `make jobs`. Should be good now.

**Comment by gcs278** [comment]
@miciah sorry, https://github.com/openshift/release/pull/57477 caused a merge conflict, and had to rebase again.

Could I get another LGTM please?

**Comment by gcs278** [comment]
Nothing changed since the last ack, and the CI job is expected to fail anyways.
/pj-rehearse ack
